modules/imap/functions.php

```diff
@@ -857,19 +857,31 @@ function imap_refresh_oauth2_token($server, $config) {
  * @return int count of messages moved
  */
 if (!hm_exists('imap_move_same_server')) {
-function imap_move_same_server($ids, $action, $hm_cache, $dest_path) {
+function imap_move_same_server($ids, $action, $hm_cache, $dest_path, $screen_emails=false) {
     $moved = array();
     $keys = array_keys($ids);
     $server_id = array_pop($keys);
     $cache = Hm_IMAP_List::get_cache($hm_cache, $server_id);
     $imap = Hm_IMAP_List::connect($server_id, $cache);
     foreach ($ids[$server_id] as $folder => $msgs) {
         if (imap_authed($imap) && $imap->select_mailbox(hex2bin($folder))) {
-            if ($imap->message_action(mb_strtoupper($action), $msgs, hex2bin($dest_path[2]))) {
+            if ($screen_emails) {
                 foreach ($msgs as $msg) {
                     $moved[]  = sprintf('imap_%s_%s_%s', $server_id, $msg, $folder);
+                    $email = current(array_column(process_address_fld($imap->get_message_headers($msg)['From']), "email"));
+                    $uid = $imap->search('ALL', false, array(array('FROM', $email)));
+                    if ($imap->message_action(mb_strtoupper($action), $uid, hex2bin($dest_path[2]))) {
+                        $moved[]  = sprintf('imap_%s_%s_%s', $server_id, $msg, $folder);
```

Comment:
I think $uid must be looped and $msg replaced with single uid values. Otherwise, you can just show a summary of the number of moved messages. In any case, it seems a bit dangerous to move all emails from a certain sender at once - there could be thousands.

Reply:
I see that fear comes back to complexity, what do you think? Should we leave this way of doing things (move all messages) or remove it?

Reply:
If you leave this code, please address the message suggestion I added above.

Reply:
Noted

Reply:
> If you leave this code, please address the message suggestion I added above.

Suggestion applied

Reply:
I don't see it. moved array should contain the msg uids of the actual moved messages. When you bulk move many messages, you only add the original msg uid and not the uids of the moved messages. This is a bug.